plugin/src/main/java/org/opensearch/ml/action/connector/DeleteConnectorTransportAction.java

```diff
@@ -58,92 +66,163 @@ public DeleteConnectorTransportAction(
-            if (Boolean.TRUE.equals(x)) {
-                try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-                    SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX);
-                    SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
-                    sourceBuilder.query(QueryBuilders.matchQuery(MLModel.CONNECTOR_ID_FIELD, connectorId));
-                    searchRequest.source(sourceBuilder);
-                    client.search(searchRequest, ActionListener.runBefore(ActionListener.wrap(searchResponse -> {
-                        SearchHit[] searchHits = searchResponse.getHits().getHits();
-                        if (searchHits.length == 0) {
-                            deleteConnector(deleteRequest, connectorId, actionListener);
-                        } else {
-                            log
-                                .error(
-                                    searchHits.length + " models are still using this connector, please delete or update the models first!"
-                                );
-                            List<String> modelIds = new ArrayList<>();
-                            for (SearchHit hit : searchHits) {
-                                modelIds.add(hit.getId());
-                            }
-                            actionListener
-                                .onFailure(
-                                    new OpenSearchStatusException(
-                                        searchHits.length
-                                            + " models are still using this connector, please delete or update the models first: "
-                                            + Arrays.toString(modelIds.toArray(new String[0])),
-                                        RestStatus.CONFLICT
-                                    )
-                                );
-                        }
-                    }, e -> {
-                        if (e instanceof IndexNotFoundException) {
-                            deleteConnector(deleteRequest, connectorId, actionListener);
-                            return;
-                        }
-                        log.error("Failed to delete ML connector: " + connectorId, e);
-                        actionListener.onFailure(e);
-                    }), context::restore));
-                } catch (Exception e) {
-                    log.error(e.getMessage(), e);
-                    actionListener.onFailure(e);
+        String tenantId = mlConnectorDeleteRequest.getTenantId();
+
+        if (mlFeatureEnabledSetting.isMultiTenancyEnabled() && Objects.isNull(tenantId)) {
+            actionListener
+                .onFailure(new OpenSearchStatusException("You don't have permission to access this connector", RestStatus.FORBIDDEN));
+            return;
+        }
```

Comment:
This would be a common code for all transport action? can it be moved to a parent or util class?